src/sonic-config-engine/portconfig.py

```diff
@@ -51,4 +70,98 @@ def parse_port_config_file(port_config_file):
             port_alias_map[data['alias']] = name
     return (ports, port_alias_map)
 
+def parse_platform_json_file(port_config_file):
+    ports = {}
+    port_alias_map = {}
+
+    # Read 'platform.json' file
+    try:
+        with open(port_config_file) as fp:
+            try:
+                data = json.load(fp)
+            except json.JSONDecodeError:
+                print("Json file does not exist")
+        port_dict = ast.literal_eval(json.dumps(data))
+    except:
+        print("error occurred while parsing json:", sys.exc_info()[1])
 
+    """ Regex for  SYMMETRIC breakout_modes
+            Syntax  : number x speed1 [speed2, speed3]
+                    : means the parent port lanes are split equally to number of ports denoted by the number. And the speed for each breakout port is default to speed1, but is changeable to speed2, speed3 etc.
+    """
+    SYMMETRIC_SPLIT_BREAKOUT_PATTERN = "(^\d{1})x(\d{2,3}G)(\[\d{2,3}G\])?"
+
+    """ Regex for ASYMMETRIC breakout_modes
+            Syntax  : number1 x speed1[speed2,..](numberA) + number2 x speedX[speedY,..](numberB) + ...
+                    : means numberA of lanes were used for number1 of port(s) with default speed1 and changeable to speed2 etc., the numberA lanes are assigned equally to the number1 of ports. Same thing for number2, speedX[speedY,..] and numberB and so on.
+    """
+    ASYMMETRIC_SPLIT_BREAKOUT_PATTERN = "(^\d{1})x(\d{2,3}G)(\[\d{2,3}G\])?(\(\d{1}\)?)\+(\d{1})x(\d{2,3}G)(\[\d{2,3}G\])?(\(\d{1}\)?)"
+
+    for interface_name in port_dict.keys():
+        parent_interface_index = re.search("Ethernet(\d+)", interface_name)
+        if parent_interface_index is not None:
+            index = int(parent_interface_index.group(1))
+
+        breakout_mode = port_dict[interface_name]["default_brkout_mode"]
+        total_lanes, alias_at_lanes = port_dict[interface_name]['lanes'], port_dict[interface_name]['alias_at_lanes']
+        total_lanes_list = total_lanes.split(",")
+        total_alias_list = [alias.strip() for alias in alias_at_lanes.split(",")]
+
+        if re.search("\+",breakout_mode) is not None:
+            intf_name_list = []
+            breakout_mode_start_index = re.search(ASYMMETRIC_SPLIT_BREAKOUT_PATTERN, breakout_mode)
+            if breakout_mode_start_index is not None:
+
+                """ first part of port configuration for default breakout mode in Asymmetric form """
+                asymmetric_breakout_mode_start_index, used_lane = int(breakout_mode_start_index.group(1)), int(breakout_mode_start_index.group(4).strip('()'))
+                asymmetric_speed = breakout_mode_start_index.group(2)
+                child_port = used_lane/asymmetric_breakout_mode_start_index
+                start, end = 0, child_port
+                for intf_name in range(index, index+used_lane, child_port):
+                    intf_name_list.append(intf_name)
+                    full_intf_name = "Ethernet"+str(intf_name)
```

Comment:
make the "Ethernet" string as a constant definition in a common place, we could change this in the future.

Reply:
updated as requested 👍